homeassistant/components/analytics/analytics.py

```diff
@@ -0,0 +1,209 @@
+"""Analytics helper class for the analytics integration."""
+import asyncio
+from typing import List, Optional
+
+import aiohttp
+import async_timeout
+
+from homeassistant.components.api import ATTR_INSTALLATION_TYPE
+from homeassistant.components.automation.const import DOMAIN as AUTOMATION_DOMAIN
+from homeassistant.components.hassio.const import DOMAIN as HASSIO_DOMAIN
+from homeassistant.components.hassio.handler import HassIO
+from homeassistant.const import __version__ as HA_VERSION
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.storage import Store
+from homeassistant.helpers.system_info import async_get_system_info
+from homeassistant.loader import async_get_integration
+
+from .const import (
+    ANALYTICS_ENDPOINT_URL,
+    ATTR_ADDON_COUNT,
+    ATTR_ADDONS,
+    ATTR_AUTO_UPDATE,
+    ATTR_AUTOMATION_COUNT,
+    ATTR_DIAGNOSTICS,
+    ATTR_HEALTHY,
+    ATTR_HUUID,
+    ATTR_INTEGRATION_COUNT,
+    ATTR_INTEGRATIONS,
+    ATTR_ONBOARDED,
+    ATTR_PREFERENCES,
+    ATTR_PROTECTED,
+    ATTR_SLUG,
+    ATTR_STATE_COUNT,
+    ATTR_SUPERVISOR,
+    ATTR_SUPPORTED,
+    ATTR_USER_COUNT,
+    ATTR_VERSION,
+    INGORED_DOMAINS,
+    LOGGER,
+    STORAGE_KEY,
+    STORAGE_VERSION,
+    AnalyticsPreference,
+)
+
+DEFAULT_DATA = {ATTR_ONBOARDED: False, ATTR_PREFERENCES: []}
+
+
+class Analytics:
+    """Analytics helper class for the analytics integration."""
+
+    def __init__(self, hass: HomeAssistant) -> None:
+        """Initialize the Analytics class."""
+        self.hass: HomeAssistant = hass
+        self.session = async_get_clientsession(hass)
+        self._data = DEFAULT_DATA
+        self._store: Store = hass.helpers.storage.Store(STORAGE_VERSION, STORAGE_KEY)
+        self._supervisor: Optional[HassIO] = self.hass.data.get(HASSIO_DOMAIN)
```

Comment:
We need a public hassio integration function that either provides this instance or create one function per api endpoint. `hass.data` is internal to each integration.